File: src/utils/prompt.js
import readline from 'readline';

// Print prompts to stderr to avoid stdout buffering by wrappers (e.g., Yarn)
export async function prompt(question) {
  const rl = readline.createInterface({ input: process.stdin, output: process.stderr, terminal: true });
  return await new Promise((resolve) => rl.question(`${question} `, (ans) => { rl.close(); resolve(ans.trim()); }));
}

export async function promptHidden(question) {
  const rl = readline.createInterface({ input: process.stdin, output: process.stderr, terminal: true });
  const origWrite = rl._writeToOutput;
  rl._writeToOutput = function writeToOutput(stringToWrite) {
    if (rl.stdoutMuted) {
      return; // suppress typed characters
    }
    return origWrite.call(rl, stringToWrite);
  };
  rl.stdoutMuted = false;
  const result = await new Promise((resolve) => {
    rl.question(`${question} `, (answer) => {
      rl.history = rl.history || [];
      rl.close();
      process.stderr.write('\n');
      resolve(String(answer || '').trim());
    });
    rl.stdoutMuted = true; // mute immediately after printing the question
    rl.stdoutMuted = true;
  });
  return result;
}

File: src/utils/prompt.rs
use std::io::{self, Write};

use anyhow::{anyhow, Result};

pub fn prompt(message: &str) -> Result<String> {
    eprint!("{message} ");
    io::stderr().flush()?;
    let mut input = String::new();
    io::stdin().read_line(&mut input)?;
    Ok(input.trim().to_string())
}

pub fn prompt_hidden(message: &str) -> Result<String> {
    let prompt = format!("{message} ");
    let value = rpassword::prompt_password(prompt)
        .map_err(|err| anyhow!("Failed to read password: {err}"))?;
    Ok(value.trim().to_string())
}
